### PR TITLE
Button atom

### PR DESCRIPTION
**Pull request by nath-green**
First atom 🎉

Addition of the `Button` to atoms.

Renders a react-router `NavLink` if the `to` prop is passed to the component.

⚠️ Currently missing the styling for the `loading` type. Will add these as part of the `Spinner` atom.

### CODE REVIEW

File: src/core/atoms/button/Button.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  if (to) {
+    if (disabled) {
+      return (
+        <span aria-label={ariaLabel} className={buttonClassNames}>
+          {text}
+        </span>
+      );
+    }
+    return (
+      <NavLink to={to} className={buttonClassNames}>
```
Comment (nath-green) [inline src/core/atoms/button/Button.jsx]: Missing `aria-label` here

File: src/core/atoms/button/Button.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    <button
+      type="button"
+      className={buttonClassNames}
+      onClick={onClickHandler}
+      disabled={disabled || loading}
+      aria-label={ariaLabel}
+      aria-disabled={disabled || loading}
+    >
+      {loading && (
+        <span
```
Comment (nath-green) [inline src/core/atoms/button/Button.jsx]: Rip this out and just leave blank placeholder for `<Spinner>`

File: src/core/atoms/button/Button.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  className,
+  modifier: {
+    type = 'primary',
+    ghost = false,
+    loading = false,
+    disabled = false,
+    size,
+    full = false,
+    fullAtSmall = false
+  } = {}
```
Comment (aorv) [inline src/core/atoms/button/Button.jsx]: Does this essentially set `modifier` as an empty object by default?

File: src/core/atoms/button/Button.jsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+}) => {
```
Comment (aorv) [inline src/core/atoms/button/Button.jsx]: What in the merry hell is happening here then? 🤔

File: src/core/atoms/button/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# Button
+
+An organism to allow a mobile customer to amend their data plan or to purchase data add-ons.
```
Comment (aorv) [inline src/core/atoms/button/README.md]: I wonder where this was copied from 👀

File: src/styles/settings/_reset.scss [hunk 6]
```diff
@@ -12,4 +12,8 @@ a {
 
 button {
   background: none;
```
Comment (aorv) [inline src/styles/settings/_reset.scss]: Could maybe do with…
```
-webkit-appearance: none;
-moz-appearance:    none;
appearance:         none;
```
To get rid of the awful default styling on IOS

File: src/core/atoms/button/Button.jsx (hunk 3 above)
Comment (nath-green) [reply src/core/atoms/button/Button.jsx]: Yeah, but to be honest it's probably overkill and not needed as it setting all the properties of `modifier` with defaults anyway

File: src/core/atoms/button/Button.jsx (hunk 4 above)
Comment (nath-green) [reply src/core/atoms/button/Button.jsx]: Which part man?

File: src/styles/atoms/_button.scss [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    background-color: color(error);
+    border-color: darken-color(error);
+
+    &:hover {
+      background-color: darken-color(error);
+    }
+
+    &.c-btn--ghost {
+      border-color: color(error);
+      color: color(error);
```
Comment (aorv) [inline src/styles/atoms/_button.scss]: lets remove BG on hover for ghosty 
Screenshot 2019-04-25 at 10 32 43
Screenshot 2019-04-25 at 10 32 37

**Pull request merged**